### PR TITLE
Allow individualized work-precision tolerances and lengths

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: src/benchmark.jl [hunk 1]
```diff
@@ -244,17 +244,13 @@ function WorkPrecisionSet(prob,
...
-                                 name=names[i],kwargs...,setups[i]...)
-    else
-      wps[i] = WorkPrecision(prob,setups[i][:alg],abstols,reltols;
-                                 appxsol=appxsol,
-                                 error_estimate=error_estimate,
-                                 name=names[i],kwargs...,setups[i]...)
-    end
+    _abstols = get(setups[i],:abstols,abstols)
+    _reltols = get(setups[i],:reltols,reltols)
+        _dts = get(setups[i],:dts,nothing)
```
Comment (YingboMa) [inline src/benchmark.jl:249]:
```suggestion
     dts = get(setups[i],:dts,nothing)
```

File: src/benchmark.jl [hunk 2]
```diff
@@ -272,36 +268,26 @@ end
...
-  else
-    sol = solve(_prob,setups[1][:alg];
-          kwargs...,setups[1]...,abstol=abstols[1],
-          reltol=reltols[1],dt=setups[1][:dts][1],
-          timeseries_errors=false,
-          dense_errors = false)
-  end
+  _abstols = get(setups[1],:abstols,abstols)
+  _reltols = get(setups[1],:reltols,reltols)
+  _dts = get(setups[1],:dts,zeros(length(_abstols)))
```
Comment (YingboMa) [inline src/benchmark.jl:273]: Is zero `dt` the sentinel value for nonspecified `dt`?

File: test/benchmark_tests.jl [hunk 3]
```diff
@@ -72,13 +72,18 @@ abstols = 1 ./10 .^(3:7)
 reltols = 1 ./10 .^(0:4)
 
 setups = [Dict(:alg=>DP5())
-          Dict(:alg=>Tsit5())]
+          Dict(:alg=>Tsit5(),
+		  :abstols => 1 ./10 .^(4:7),
```
Comment (YingboMa) [inline test/benchmark_tests.jl:76]: Tab?

File: src/benchmark.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/benchmark.jl:249]: It should be `_dts`?

File: src/benchmark.jl (hunk 2 above)
Comment (ChrisRackauckas) [reply src/benchmark.jl:273]: yes

File: src/benchmark.jl (hunk 1 above)
Comment (YingboMa) [reply src/benchmark.jl:249]: Oops, I meant to comment on the indentation. :P

File: test/benchmark_tests.jl [hunk 4]
```diff
@@ -73,8 +73,8 @@ reltols = 1 ./10 .^(0:4)
 
 setups = [Dict(:alg=>DP5())
           Dict(:alg=>Tsit5(),
-		  :abstols => 1 ./10 .^(4:7),
-		  :reltols => 1 ./10 .^(1:4))]
+		  	   :abstols => 1 ./10 .^(4:7),
```
Comment (YingboMa) [inline test/benchmark_tests.jl:76]: Oh no 🙀 it seems like there are 3 tabs. We should only use spaces, right?

Comment (YingboMa) [reply test/benchmark_tests.jl:76]: Is GitHub playing tricks on me? They look like tabs to me.

**Comment by YingboMa** [comment]
`get` looks so much nicer!

Is there way in Julia to say “use the default kwarg”? Constructing default kwarg seems like a pretty common thing to do, yet error prone.

**Pull request merged**